packages/drafts-realm/ai-app-generator.gts

```diff
@@ -0,0 +1,452 @@
+import { type CardDef } from 'https://cardstack.com/base/card-api';
+import {
+  CardContainer,
+  FieldContainer,
+  BoxelInput,
+  Button,
+} from '@cardstack/boxel-ui/components';
+import { cssVar, eq, getContrastColor } from '@cardstack/boxel-ui/helpers';
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import GlimmerComponent from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { AppCard, Tab } from './app-card';
+
+const getComponent = (card: CardDef) => {
+  if (!card) {
+    return;
+  }
+  return card.constructor.getComponent(card);
+};
+
+class Requirements extends GlimmerComponent<{
+  instances: CardDef[] | [];
+}> {
+  <template>
+    <div class='requirements'>
+      <aside class='recent-reqs-sidebar'>
+        <h3 class='recent-reqs-title'>Recent Requirements</h3>
+        <ul>
+          {{#each @instances as |doc|}}
+            <li>
+              <button {{on 'click' (fn this.openDoc doc)}}>
+                {{doc.title}}
+              </button>
+            </li>
+          {{/each}}
+        </ul>
+      </aside>
+      {{#let (getComponent this.currentDoc) as |Card|}}
+        <div class='requirements-doc'>
+          <Card />
+        </div>
+      {{/let}}
+    </div>
+    <style>
+      .requirements {
+        height: 100%;
+        display: grid;
+        grid-template-columns: auto 1fr;
+        gap: var(--boxel-sp);
+      }
+      .recent-reqs-sidebar {
+        max-width: 235px;
+      }
+      .recent-reqs-title {
+        margin: 0;
+        font: 700 var(--boxel-font);
+      }
+      .requirements-doc > :deep(.field-component-card.embedded-format) {
+        --overlay-embedded-card-header-height: 0;
+      }
+    </style>
+  </template>
+
+  @tracked _doc?: CardDef;
+
+  @action openDoc(doc: CardDef) {
+    this._doc = doc;
+  }
+
+  get currentDoc() {
+    return this._doc ?? this.args.instances?.[0];
+  }
+}
+
+class Isolated extends AppCard.isolated {
+  <template>
+    <section class='app'>
+      <header
+        class='app-header'
+        style={{cssVar
+          header-background-color=this.headerColor
+          header-text-color=(if
+            this.headerColor (getContrastColor this.headerColor)
+          )
```

Comment:
Is this conditional needed? I thought `getContrastColor` returns undefined if given undefined?

Reply:
background-color was getting set to 'null' for one of them. I updated it so it's undefined and this conditional is not needed.